apiserver/remoterelations/remoterelations.go

```diff
@@ -460,38 +478,46 @@ func (api *RemoteRelationsAPI) registerRemoteRelation(relation params.RegisterRe
 	})
 	// If it already exists, that's fine.
 	if err != nil && !errors.IsAlreadyExists(err) {
-		return errors.Annotatef(err, "adding remote application %v", uniqueRemoteApplicationName)
+		return nil, errors.Annotatef(err, "adding remote application %v", uniqueRemoteApplicationName)
 	}
-	logger.Debugf("added remote application %v to local model", uniqueRemoteApplicationName)
+	logger.Debugf("added remote application %v to local model with token %v", uniqueRemoteApplicationName, relation.ApplicationId.Token)
 
 	// Now add the relation if it doesn't already exist.
 	localRel, err := api.st.EndpointsRelation(*localEndpoint, remoteEndpoint)
 	if err != nil && !errors.IsNotFound(err) {
-		return errors.Trace(err)
+		return nil, errors.Trace(err)
 	}
 	if err != nil {
 		localRel, err = api.st.AddRelation(*localEndpoint, remoteEndpoint)
 		// Again, if it already exists, that's fine.
 		if err != nil && !errors.IsAlreadyExists(err) {
-			return errors.Annotate(err, "adding remote relation")
+			return nil, errors.Annotate(err, "adding remote relation")
 		}
 		logger.Debugf("added relation %v to model %v", localRel.Tag().Id(), api.st.ModelUUID())
 	}
 
 	// Ensure we have references recorded.
-	logger.Debugf("importing remote application and relation into model %v", api.st.ModelUUID())
+	logger.Debugf("importing remote relation into model %v", api.st.ModelUUID())
 	logger.Debugf("remote model is %v", remoteModelTag.Id())
-	err = api.st.ImportRemoteEntity(remoteModelTag, names.NewApplicationTag(uniqueRemoteApplicationName), relation.ApplicationId.Token)
```

Comment:
It's done as part of creating the remote application. The AddRemoteApplicationParams struct has a token attribute. So essentially we were attempting to import twice. The second time was here and it failed with an AlreadyExists error so the error was ignored. But we didn't need to do it twice.